### DISCUSSION TITLE
How to update GeneratedIconSource.Text in C# code after the 2.0.99 release?

### DISCUSSION THREAD

**Discussion by myfix16** [HavenDV/H.NotifyIcon#83] [Q&A] [answered]
Hi, I noticed the `GeneratedIcon` class was replaced by `GeneratedIconSource` in the 2.0.99 release, which was an incompatible API change. I used to change `GeneratedIcon.Text` programmatically using 
```
GeneratedIcon icon = trayIcon.GeneratedIcon;
icon.Text = "new text";
```
Now I have updated the code to 
```
var icon = (GeneratedIconSource)trayIcon.IconSource;
icon.Text = newPercentText;
```
, but it has no effect. 

I'm using H.NotifyIcon.WinUI. And I wonder what's the correct way to update `GeneratedIconSource.Text` (or other properties) in C# code.

**Comment by myfix16** [comment, marked as answer]
I figured out the solution. In this version, you need to call `TaskbarIcon.UpdateIcon(GeneratedIcon.ToIcon())` to update the icon manually.
 
@HavenDV I suppose the update should be done automatically. So I guess the following method in `TaskbarIcon.IconSource.cs` is not called when a property of `GeneratedIcon` changes:
```
async partial void OnIconSourceChanged(ImageSource? oldValue, ImageSource? newValue)
    {
        if (newValue == null)
        {
            Icon = null;
            return;
        }

#if MACOS
        TrayIcon.Icon = NSImage.FromStream(stream);
#else
        Icon = await newValue.ToIconAsync().ConfigureAwait(true);
#endif
    }
```

**Reply by HavenDV** [reply]
I'm sorry I missed your message somehow. Thanks for the research, I'll check it out. I think it was right to be in bugs

**Reply by HavenDV** [reply]
Please check latest version

**Reply by myfix16** [reply]
@HavenDV I encountered an error when installing the latest nuget package: Cannot find package `H.GeneratedIcons.System.Drawing`
![image](https://user-images.githubusercontent.com/61649477/228175076-7cfd3854-17a1-4e8a-b404-3d58be7e93e4.png)

**Reply by HavenDV** [reply]
Sorry, I forgot to release them. Will be available in about 10 minutes
